hatchery/pods.go

```diff
@@ -79,35 +92,49 @@ func statusK8sPod(userName string) (*WorkspaceStatus, error) {
 		status.Status = "Stopped"
 		return &status, nil
 	case "Pending":
-		status.Status = "Launching"
-		return &status, nil
 	case "Running":
-		break
-	default:
-		fmt.Printf("Unknown pod status for %s: %s\n", podName, string(pod.Status.Phase))
-	}
+		var allReady = true
+		if pod.Status.Phase == "Pending" {
+			allReady = false
+		} else {
+			for _, v := range pod.Status.Conditions {
+				if v.Type == "Ready" {
+					if v.Status != "True" {
+						allReady = false
+					}
+				} else if v.Type == "PodScheduled" {
+					if v.Status != "True" {
+						allReady = false
+					}
+				}
+			}
+		}
```

Comment:
I think this works as is, but might be more readable/testable as a separate func

```go
func getPodStatus(pod corev1.CoreV1Interface) {
    if pod.Status.Phase == "Pending" {
        return false
    }
    for _, v := range pod.Status.Conditions {
        if (v.Type == "Ready" || v.Type == "PodScheduled") && v.Status != "True" {
            return false
        }
    }
    return true
}
```